File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0) return false;
        int n = x;
        int temp = 0;

        while(n != 0) {
            temp = temp*10 + n%10;
            n = n/10;
        }

        return temp == x;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) {
            return false;
        }

        int revNum = 0; 
        int copy = x;
        while(x != 0) {
            int dig = x%10;
            if(revNum > INT_MAX/10 || revNum < INT_MIN/10) return false;
            revNum = revNum*10 + dig;
            x = x/10;
        }  
        return copy == revNum;    
    }
};